packages/token-search-discovery-controller/src/token-search-api-service/token-search-api-service.ts

```diff
@@ -0,0 +1,48 @@
+import { AbstractTokenSearchApiService } from './abstract-token-search-api-service';
+import type { TokenSearchParams, TokenSearchResponseItem } from '../types';
+
+export class TokenSearchApiService extends AbstractTokenSearchApiService {
+  readonly #baseUrl: string;
+
+  constructor(baseUrl: string) {
+    super();
+    if (!baseUrl) {
+      throw new Error('Portfolio API URL is not set');
+    }
+    this.#baseUrl = baseUrl;
+  }
+
+  async searchTokens(
+    tokenSearchParams?: TokenSearchParams,
+  ): Promise<TokenSearchResponseItem[]> {
+    const queryParams = new URLSearchParams();
+
+    if (tokenSearchParams?.chains && tokenSearchParams.chains.length > 0) {
+      queryParams.append('chains', tokenSearchParams.chains.join());
+    }
+    if (tokenSearchParams?.name) {
+      queryParams.append('name', tokenSearchParams.name);
+    }
+    if (tokenSearchParams?.limit) {
+      queryParams.append('limit', tokenSearchParams.limit);
+    }
+
+    const queryString = queryParams.toString();
+    const url = `${this.#baseUrl}/tokens-search/name${queryString ? `?${queryString}` : ''}`;
```

Comment:
Instead of appending the query string manually what are your thoughts on using the URL constructor?

``` suggestion
    const url = new URL('/tokens-search/name', this.#baseUrl);
    
    if (tokenSearchParams?.chains && tokenSearchParams.chains.length > 0) {
      url.searchParams.append('chains', tokenSearchParams.chains.join());
    }
    if (tokenSearchParams?.name) {
      url.searchParams.append('name', tokenSearchParams.name);
    }
    if (tokenSearchParams?.limit) {
      url.searchParams.append('limit', tokenSearchParams.limit);
    }

Reply:
You're probably right on this point.  It's more readable and should take care of the case when no params are provided and I had to use the ternary.  I'll work on it.